### PR TITLE
[7.7.x] RHPAM-540 (JBPM-7031): Stunner - Disable doing operation in view mode

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/RHPAM-540

This is the corollary of https://github.com/kiegroup/kie-wb-common/pull/1654 for 7.7.x

(cherry picked from commit 5666b305e964f55fce1cd5bac4af65d89d9932ec)

### CODE REVIEW

File: kie-wb-common-stunner/kie-wb-common-stunner-extensions/kie-wb-common-stunner-project/kie-wb-common-stunner-project-client/src/main/java/org/kie/workbench/common/stunner/project/client/editor/AbstractProjectDiagramEditor.java [hunk 1]
```diff
@@ -256,6 +302,70 @@ public void onError(final ClientRuntimeError error) {
...
+        switchGridItem.setEnabled(enabled);
+        validateItem.setEnabled(enabled);
+        exportsItem.setEnabled(enabled);
+
+        deleteSelectionItem.setEnabled(false);
+        undoItem.setEnabled(false);
+        redoItem.setEnabled(false);
+        copyItem.setEnabled(false);
+        cutItem.setEnabled(false);
+        pasteItem.setEnabled(false);
```
Comment (jomarko) [inline kie-wb-common-stunner/kie-wb-common-stunner-extensions/kie-wb-common-stunner-project/kie-wb-common-stunner-project-client/src/main/java/org/kie/workbench/common/stunner/project/client/editor/AbstractProjectDiagramEditor.java]: Not sure why didn't asked during master sibling review, however are these hardcoded `false` fine?

Comment (manstis) [reply kie-wb-common-stunner/kie-wb-common-stunner-extensions/kie-wb-common-stunner-project/kie-wb-common-stunner-project-client/src/main/java/org/kie/workbench/common/stunner/project/client/editor/AbstractProjectDiagramEditor.java]: @jomarko Yes, when the editor is first opened (in edit or read-only mode) the "delete selection", "undo", "redo", "copy", "cut" and "paste" menu bar icons should be disabled as nothing has been selected by the User at this stage. When a selection occurs there is some code [here](https://github.com/manstis/kie-wb-common/blob/08ca6edac5da3e291c485ff14a52b2d95efc46da/kie-wb-common-stunner/kie-wb-common-stunner-extensions/kie-wb-common-stunner-project/kie-wb-common-stunner-project-client/src/main/java/org/kie/workbench/common/stunner/project/client/editor/AbstractProjectDiagramEditor.java#L459) that handles them being enabled.

Comment (manstis) [reply kie-wb-common-stunner/kie-wb-common-stunner-extensions/kie-wb-common-stunner-project/kie-wb-common-stunner-project-client/src/main/java/org/kie/workbench/common/stunner/project/client/editor/AbstractProjectDiagramEditor.java]: @jomarko "undo" and "redo" state may need addressing for DMN (as the User can "undo" navigation back to the graph editor) however I have integration of the recent Stunner changes I've made (read-only, XML editor, etc) on my "return to DMN" todo list.

**Comment by jomarko** [comment]
@manstis during manual check I noticed that in view mode user can expand right side docks and put there values. These don't seem to be stored however I think we should hide/disable these docks, separate jira in the air?

**Pull request merged**